x-pack/plugins/infra/public/components/log_stream/log_stream.tsx

```diff
@@ -71,6 +71,9 @@ interface LogStreamContentProps {
   center?: LogEntryCursor;
   highlight?: string;
   columns?: LogColumnDefinition[];
+  hasColumnHeaders?: boolean;
+  loadingState?: React.ReactNode;
+  emptyState?: (options: { handleReload: () => void }) => React.ReactNode;
```

Comment:
How about we formulate both as render props for symmetry and prefix their names with `render`?

```suggestion
  renderLoadingState?: () => React.ReactNode;
  renderEmptyState?: (options: { handleReload: () => void }) => React.ReactNode;
```